src/state.rs

```diff
@@ -201,46 +239,24 @@ impl NetworkState {
         let ret = result.cast().map_err(VMError::from_store_error)?;
 
         // If we reach this point, everything went well and we can use the
-        // updates made in the forked state.
-        *self = fork;
+        // updates made in the forked state. The forked snapshot is set as
+        // the latest, and the latest is hashed and put in the map of
+        // snapshots.
+        mem::swap(&mut self.latest, &mut snap);
+
+        let snap_id = Store::hash(&snap.encode_to_vec());
+        self.snapshots
+            .insert(snap_id, snap)
+            .map_err(VMError::from_store_error)?;
 
```

Comment:
I guess similar logic to the above will be added in ops/transact for inter-contract calls

Reply:
Iḿ not sure I understand. Could you elaborate?

Reply:
there are 2 places from which we are calling transact - from state and from ops/transact. You took care of one of these places, here in state.rs, but there is one more place that changes network/contract state, see ops/transact.rs

Reply:
as discussed on Friday - the above is not valid if we assume that "transact" is just a name, and the actual meaning is more like Ethereum CALL or DELEGATECALL - in such case it is not needed to add snapshot logic in ops/transact - the name of the method is misleading though